### ISSUE TITLE
How to reset / config the vagrant winnfsd plugin?

### ISSUE THREAD

**Issue by webprogrammierer** [geerlingguy/drupal-vm#2129]
### Issue Type
Support Request

### Your Environment
Vagrant 2.2.14
VirtualBox 6.1

### Your OS
Windows 10

### Full console output

```
=====================================================
WinNFSd 2.4.0 [5f7f224]
Network File System server for Windows
Copyright (C) 2005 Ming-Yang Kao
Edited in 2011 by ZeWaren
Edited in 2013 by Alexander Schneider (Jankowfsky AG)
Edited in 2014 2015 by Yann Schepens
Edited in 2016 by Peter Philipp (Cando Image GmbH), Marc Harding
=====================================================
Path #1 is: D:/drupal-vm/drupal-vm, path alias is: /D/drupal-vm/drupal-vm
Path #2 is: D:/xxx-projekte, path alias is: /D/xxx-projekte
Path #3 is: D:/drupal-projekte/.., path alias is: /D/drupal-projekte/..
Path #4 is: D:/drupal-projekte/.., path alias is: /D/drupal-projekte/..
Path #5 is: D:/drupal-projekte.., path alias is: /D/drupal-projekte/..
Path #6 is: D:/drupal-projekte/.., path alias is: /D/drupal-projekte/..
Path #7 is: D:/drupal-projekte/.., path alias is: /D/drupal-projekte/..
Path #8 is: D:/drupal-projekte/.., path alias is: /D/drupal-projekte/..
Path #9 is: D:/drupal-projekte/.., path alias is: /D/drupal-projekte/..
Path #10 is: D:/drupal-projekte/.., path alias is: /D/drupal-projekte/..
Path #11 is: D:/drupal-projekte/.., path alias is: /D/drupal-projekte/..
Path #12 is: D:/drupal-projekte/.., path alias is: /D/drupal-projekte/..
Path D:/drupal-vm/drupal-vm with path alias /D/drupal-vm/drupal-vm already known
Path #13 is: D:/drupal-vm/drupal-vm2, path alias is: /D/drupal-vm/drupal-vm2
Path #14 is: D:/drupal-projekte/.., path alias is: /D/drupal-projekte/..
Path D:/drupal-projekte/.. with path alias /D/drupal-projekte/.. already known
Path D:/drupal-vm/drupal-vm2 with path alias /D/drupal-vm/drupal-vm2 already known
Path D:/drupal-vm/drupal-vm2 with path alias /D/drupal-vm/drupal-vm2 already known
Path D:/drupal-projekte/.. with path alias /D/drupal-projekte/.. already known
Path D:/drupal-projekte/.. with path alias /D/drupal-projekte/.. already known
Path D:/drupal-vm/drupal-vm2 with path alias /D/drupal-vm/drupal-vm2 already known
Portmap daemon started
NFS daemon started
Mount daemon started
Listening on 0.0.0.0
Type 'help' to see help

Path D:/drupal-vm/drupal-vm with path alias /D/drupal-vm/drupal-vm already known
custimecare.de already known
Path D:/drupal-vm/drupal-vm with path alias /D/drupal-vm/drupal-vm already known
Path D:/drupal-vm/drupal-vm2 with path alias /D/drupal-vm/drupal-vm2 already known
Path D:/drupal-vm/drupal-vm2 with path alias /D/drupal-vm/drupal-vm2 already known
Path D:/drupal-vm/drupal-vm with path alias /D/drupal-vm/drupal-vm already known
Path D:/drupal-vm/drupal-vm2 with path alias /D/drupal-vm/drupal-vm2 already known
Path D:/drupal-vm/drupal-vm2 with path alias /D/drupal-vm/drupal-vm2 already known
Path D:/drupal-projekte/.. with path alias /D/drupal-projekte/.. already known
... a lot of lines removed ...
... a lot of lines removed ...
... a lot of lines removed ...
... a lot of lines removed ...
Path D:/drupal-vm/drupal-vm2 with path alias /D/drupal-vm/drupal-vm2 already known
Path D:/drupal-vm/drupal-vm with path alias /D/drupal-vm/drupal-vm already known
```

### Summary

I have only 4 shared folders! All this entries above are old entries. HOW CAN I get RID of them?

Actual list command in winnfsd shell:
```
list
=====================================================
192.168.88.89
192.168.88.89
192.168.88.89
192.168.88.89
There are 4 clients mounted.
=====================================================
```

So how can I totally reset the winnfsd (vagrant plugin) to totally re-initialize, to remove all (!) entries, to set it up new and clean?
All this does NOT help and not clean up my winnfsd configuration:
vagrant halt/up, vagrant destroy/up, vagrant provision, vagrant plugin uninstall vagrant-winnfsd, vagrant plugin install ...

Nothing helps to clean my winnfsd and get an new initialized state.
All this entries above are from old vagrant ups. I want to get rid of them. How?
Thank you.

**Comment by geerlingguy** [comment]
Honestly I have not used NFS on windows in over four years, and have no idea how to get it working properly, how to configure it, etc.

The last few times I used Drupal VM on Windows I either used SMB (which was fast enough for my liking) or rsync (which was even faster but required waiting for sync to complete).

**Comment by webprogrammierer** [comment]
You do not use your own Drupal VM? Or do you use it on other operating system? Which?

So I do not understand why still today in the latest version of Drupal VM in config.yml there is "nfs" used as standard file system for vagrant_synced_folder_default_type and for vagrant_synced_folders.

And there is no hint in config.yml, that if somebody uses the VM on Windows he should rewrite all this. Maybe there should be a default-windows.config.yml file additionally ready to rename and use for Windows users.

By the way, the `drupal_composer_project_options `say that drupal should be installed in dev version. Nobody needs a Drupal in dev version. This should be changed like this

```
#drupal_composer_project_package: "drupal/recommended-project:^9@dev"   #old
#drupal_composer_project_options: "--prefer-dist --stability dev --no-interaction"  #old

drupal_composer_project_package: "drupal/recommended-project"
drupal_composer_project_options: "--prefer-dist --stability stable --no-interaction --no-dev"
```

But back to NFS.
**It would be great if somebody else could answer my question.**

**Comment by geerlingguy** [comment]
I don't use Windows.

I do use Drupal VM.

Using a dev release is standard practice for testing / building a quick demo site to work on something.

NFS is the best default for macOS, and a good default for Linux, and 'can be made to work' on Windows.

The number of devs who work on Drupal sites and work on Windows is very small (proportionately) compared to Mac/Linux devs, so a lot of the tooling you'll find works on Windows, but will often require some extra tweaks, extra setup, or extra pain as you get them set up, since it is harder to get a lot of open source software working as well on Windows.

I wrote about this a few years back here: https://www.jeffgeerling.com/blog/2019/developing-virtualbox-and-vagrant-on-windows

**Comment by webprogrammierer** [comment]
I understand, you are using a Mac :-)

But you wrote a Virtual Machine for Windows. This is a great thing. 
Maybe the config.yml could contain more comments about Windows config options.


**But back to NFS.
It would be great if somebody else could answer my question.**

**Comment by very-random-man** [comment]
I struggled with getting NFS working satisfactorily on windows. I found it quickest to either use samba mounts or rsync with the plugins that make it work better. I've used both versions of WSL too but I had the best results by creating a hyper-v Linux machine and running everything from inside that. Then it's just native Linux without any kludges and stuff like NFS just works. The networking is tricky to get your head around but i can confirm it's possible and results in a reasonable dev environment.
